libraries/botbuilder-azure-blobs/src/blobsStorage.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import getStream from 'get-stream';
+import pmap from 'p-map';
+import { ContainerClient, StoragePipelineOptions } from '@azure/storage-blob';
+import { Storage, StoreItems, isStoreItems } from 'botbuilder';
+import { ignoreError, isStatusCodeError } from './ignoreError';
+import { sanitizeBlobKey } from './sanitizeBlobKey';
+
+/**
+ * BlobsStorage provides a [Storage](xref:botbuilder-core.Storage) implementation backed by Azure Blob Storage
+ */
+export class BlobsStorage implements Storage {
+    private readonly _containerClient: ContainerClient;
+    private readonly _concurrency: number;
+    private _initializePromise: Promise<unknown>;
+
+    /**
+     * Constructs a BlobsStorage instance.
+     *
+     * @param connectionString Azure Blob Storage connection string
+     * @param containerName Azure Blob Storage container name
+     * @param options Azure Blob Storage [StoragePipelineOptions](xref:@azure/storage-blob.StoragePipelineOptions) options
+     */
+    constructor(connectionString: string, containerName: string, options?: StoragePipelineOptions) {
+        if (typeof connectionString !== 'string') {
+            throw new Error('connectionString is required and must be a string');
+        }
+
+        if (typeof containerName !== 'string') {
+            throw new Error('containerName is required and must be a string');
+        }
```

Comment:
Since you're using `typeof` you could use TypeErrors instead (personal preference).

However, these checks don't cover empty strings, is that an acceptable scenario?

Reply:
Moved to type assertions.